ecosystem/typescript/sdk/src/transaction_builder/builder.ts

```diff
@@ -257,7 +257,7 @@ export class TransactionBuilderABI {
       payload = new TransactionPayloadScript(new Script(funcABI.code, typeTags, scriptArgs));
     }
 
-    return payload;
```

Comment:
would be better to refactor the 2 if statements above into an `if, elseif, else` chain and throw in the else chain if none of the cases match. Then you don't need to do the `!` stuff below.

Reply:
Actually, this is another fair statement. The only drawback is that we have to add a test to cover the exception thrown. Otherwise, the code coverage is dropping. I would just stick with `paylaod!` as the ABI only supports EntryFunctionABI and TransactionScriptABI

Reply:
I honestly prefer a small drop in code coverage (which is some arbitrary metric anyways) over using TypeScript anti-patterns and not-so-clear code. usin ghte if, elseif else is also safer in the sense that it avoids you falling through in this function if later on another transaction type emerges.

Reply:
done